cli/src/merge_tools/builtin.rs

```diff
@@ -722,13 +722,11 @@ mod tests {
                 }
             }
         }
-        let merge = Merge::new(
-            vec![to_file_id(base_tree.path_value(&path))],
-            vec![
-                to_file_id(left_tree.path_value(&path)),
-                to_file_id(right_tree.path_value(&path)),
-            ],
-        );
+        let merge = Merge::from_vec(vec![
+            to_file_id(left_tree.path_value(&path)),
+            to_file_id(base_tree.path_value(&path)),
+            to_file_id(right_tree.path_value(&path)),
+        ]);
```

Comment:
Is the `vec!` needed or can we pass an array? (Question applies throughout the commit)

Reply:
`vec!` is needed (unless the array length is 1), but the allocated buffer will be reused.
https://docs.rs/smallvec/latest/smallvec/struct.SmallVec.html#impl-From%3CVec%3C%3CA+as+Array%3E::Item,+Global%3E%3E-for-SmallVec%3CA%3E

We could add `from_iter()`, but that would go against the design of `MergeBuilder`. I don't think it's good idea to create `SmallVec` by caller either. That's why `values` is `impl Into<SmallVec<..>>`.